source/__tests__/integration/create-resource.test.js

```diff
@@ -154,15 +157,15 @@ describe('Create profile resource template requirements', () => {
           propertyURI: 'http://www.example.org',
           propertyLabel: 'propLabel'
         })
-        // wait for resourceURI check
+        // wait for resourceURI and propertyURI checks
+        await page.waitFor(1000, {waitUntil: 'networkidle2'})
         let valid_url_class = await page.$('input[name="resourceURI"]', e => e.getAttribute('ng-valid-url'))
         expect(valid_url_class).toBeTruthy()
-        // wait for propertyURI check
         valid_url_class = await page.$('input[name="propertyURI"]', e => e.getAttribute('ng-valid-url'))
         expect(valid_url_class).toBeTruthy()
 
         await page.click(exportButtonSel)
-        await page.waitForSelector('a[download="My profile.json"]')
+        await page.waitForSelector('a[download="My profile.json"]', {visible: true})
```

Comment:
the `visible: true` option helps here, but it seemed to time out in some other places I tried it.  🤷‍♀

Reply:
maybe the element in question never actually showed up?  bad selector in the cases where it didn't work?  or do you mean e.g., it worked without `visible: true`, but failed with it?

Reply:
i no longer remember.  I tried ... so ... many ... things ...